python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1209,3 +1213,100 @@ def empty_subset(
         cls, partitions_def: Optional[PartitionsDefinition] = None
     ) -> "DefaultPartitionsSubset":
         return cls()
+
+
+class AllPartitionsSubset(
+    NamedTuple(
+        "_AllPartitionsSubset",
+        [
+            ("partitions_def", PartitionsDefinition),
+            ("instance_queryer", "CachingInstanceQueryer"),
+        ],
+    ),
+    PartitionsSubset,
+):
+    """This is an in-memory (i.e. not serializable) convenience class that represents all partitions
+    of a given PartitionsDefinition, allowing set operations to be taken without having to load
+    all partition keys immediately.
+    """
+
+    def get_partition_keys(self, current_time: Optional[datetime] = None) -> Sequence[str]:
+        return self.partitions_def.get_partition_keys(current_time, self.instance_queryer)
+
+    def get_partition_keys_not_in_subset(
+        self,
+        partitions_def: PartitionsDefinition,
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> Iterable[str]:
+        return set()
+
+    def get_partition_key_ranges(
+        self,
+        partitions_def: PartitionsDefinition,
+        current_time: Optional[datetime] = None,
+        dynamic_partitions_store: Optional[DynamicPartitionsStore] = None,
+    ) -> Sequence[PartitionKeyRange]:
+        first_key = partitions_def.get_first_partition_key(current_time, dynamic_partitions_store)
```

Comment:
nit: similar to other comment, make current time and dynamic partitions store args accept the `self.instance_queryer.evaluation_time` and `self.instance_queryer` respectively

Reply:
These ones were intentionally left as they are, with the thinking being that it'd be confusing if passing in current_time did nothing in these cases.

However, that made me realize that it'd make more sense to do what you're suggesting, then error if a current time is passed in -- thoughts on that approach?

Reply:
yes, that sounds good